File: yizhi/Alien-V4-improved-master/src/main/java/dev/gzsakura_miitong/asm/accessors/IVec3d.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  net.minecraft.util.math.Vec3d
 *  org.spongepowered.asm.mixin.Mixin
 *  org.spongepowered.asm.mixin.Mutable
 *  org.spongepowered.asm.mixin.gen.Accessor
 */
package dev.gzsakura_miitong.asm.accessors;

import net.minecraft.util.math.Vec3d;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.Mutable;
import org.spongepowered.asm.mixin.gen.Accessor;

@Mixin(value={Vec3d.class})
public interface IVec3d {
    @Mutable
    @Accessor(value="x")
    public void setX(double var1);

    @Mutable
    @Accessor(value="y")
    public void setY(double var1);

    @Mutable
    @Accessor(value="z")
    public void setZ(double var1);
}

File: src/main/kotlin/com/zychen027/MeteorPlusPlus/utils/efa/asm/IVec3d.kt
package com.zychen027.meteorplusplus.utils.efa.asm

import net.minecraft.util.math.Vec3d
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.Mutable
import org.spongepowered.asm.mixin.gen.Accessor

@Mixin(Vec3d::class)
interface IVec3d {
    @Mutable
    @Accessor("x")
    fun setX(x: Double)

    @Mutable
    @Accessor("y")
    fun setY(y: Double)

    @Mutable
    @Accessor("z")
    fun setZ(z: Double)
}
